src/Uno.UI/UI/Xaml/Controls/ScrollContentPresenter/ScrollContentPresenter.Native.cs

```diff
@@ -7,12 +7,31 @@ namespace Windows.UI.Xaml.Controls
 {
 	// This file only contains support of NativeScrollContentPresenter
 
-	partial class ScrollContentPresenter
+	partial class ScrollContentPresenter : IFrameworkTemplatePoolAware
 	{
 		internal INativeScrollContentPresenter Native { get; set; }
 
 		private object RealContent => Native?.Content;
 
+		public void OnTemplateRecycled()
+		{
+			if (TemplatedParent is null && Native is { })
+			{
+				Native.Content = null;
+				Native = null;
+			}
+		}
+
+		protected internal override void OnTemplatedParentChanged(DependencyPropertyChangedEventArgs e)
+		{
+			if (e.NewValue is null)
```

Comment:
```suggestion
            // Clear the native content, in case this instance is being explicitly removed from its templated parent.
            // Note that in case the template is being recycled, the parent is not explicitly removed, but rather
            // relies on weak references, and this method is not called.
			if (e.NewValue is null)
```